### PR TITLE
Add custom token ui2

### PR DESCRIPTION
**Pull request by dbo1dyrev**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

Screenshot 2021-07-15 at 23 25 04
Screenshot 2021-07-15 at 21 31 43
Screenshot 2021-07-15 at 23 03 31
Screenshot 2021-07-15 at 23 03 42
Screenshot 2021-07-15 at 23 03 59

https://user-images.githubusercontent.com/68349436/125850702-16e2a643-96f4-4a3b-95a4-a4c629fc6f33.mov

https://user-images.githubusercontent.com/68349436/126489051-669d9de3-bea4-4297-8001-78fd7d5eeff0.mov

### CODE REVIEW

**Comment by dbo1dyrev** [comment]
Additional info: without autocomplete api you can't see this modal

File: src/components/scenes/AddTokenScene.js [hunk 1]
```diff
@@ -35,199 +34,224 @@ type DispatchProps = {
...
+      const denomination = decimalPlacesToDenomination(decimalPlaces)
+
+      const isExists = currencyInfos.some(
+        (item: any) =>
+          item.currencyCode === currencyCode ||
+          item.currencyName === currencyName ||
+          item.decimalPlaces === decimalPlaces ||
+          item.contractAddress === contractAddress
+      )
+
```
Comment (thehobbit85) [inline src/components/scenes/AddTokenScene.js:122]: Move to utility function

File: src/components/scenes/AddTokenScene.js [hunk 2]
```diff
@@ -35,199 +34,224 @@ type DispatchProps = {
...
+const DECIMAL_PLACES_MAX_LENGTH = 1
 
-  onChangeDecimalPlaces = (input: string) => {
-    this.setState({
-      decimalPlaces: input
-    })
-  }
+const currencyCodeValidation = (value: string): string => (/[^a-z\d]/i.test(value) ? s.strings.warning_alphanumeric : '')
+
+export const AddToken = ({ addTokenPending, currentCustomTokens = [], currencyInfos, wallet, walletId, currencyWallet, addNewToken, onAddToken }: Props) => {
```
Comment (thehobbit85) [inline src/components/scenes/AddTokenScene.js:74]: Should be scrollable

File: src/components/scenes/AddTokenScene.js [hunk 3]
```diff
@@ -35,199 +34,224 @@ type DispatchProps = {
...
+          value={contractAddress}
+          returnKeyType={contractAddressReturnKeyType}
+          label={s.strings.addtoken_contract_address_input_text}
+          maxLength={CONTRACT_ADDRESS_MAX_LENGTH}
+          showSearchIcon={false}
+          autoCorrect={false}
+          marginRem={[0.5, 0.6, 1]}
+          onSubmitEditing={handleSubmitEditing}
+        />
+        <EdgeTextFieldOutlined
```
Comment (thehobbit85) [inline src/components/scenes/AddTokenScene.js:221]: Should also have validation - Should "clean" and not allow non number chars.

File: src/components/themed/Checkbox.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  onChange(value: boolean): void
+}
+
+export const Checkbox = ({ textStyle, disabled, children, value, onChange, marginRem, paddingRem, ellipsizeMode, numberOfLines }: Props) => {
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const spacings = { ...getMarginSpacingStyles(marginRem || 0.5, theme.rem), ...getPaddingSpacingStyles(paddingRem || 1, theme.rem) }
+
+  return (
+    <TouchableWithoutFeedback onPress={() => onChange(!value)} disabled={disabled}>
```
Comment (thehobbit85) [inline src/components/themed/Checkbox.js:29]: Should only be able to tap on checkbox and not the entire row

File: src/components/scenes/AddTokenScene.js [hunk 5]
```diff
@@ -35,199 +34,224 @@ type DispatchProps = {
...
+        setCurrencyName(currencyName || currencyInfo.currencyName)
+        setDecimalPlaces(decimalPlaces || currencyInfo.decimalPlaces)
+        setContractAddress(contractAddress || currencyInfo.contractAddress)
       }
-    )
+
+      const nextFocusedField = inputRefs[dataEmptyIndex].current
+
+      if (nextFocusedField !== null) nextFocusedField.focus()
+    }
```
Comment (thehobbit85) [inline src/components/scenes/AddTokenScene.js:162]: Separate logic

File: src/components/scenes/AddTokenScene.js [hunk 6]
```diff
@@ -35,199 +34,224 @@ type DispatchProps = {
...
-const styles: typeof rawStyles = StyleSheet.create(rawStyles)
+}))
 
 export const AddTokenScene = connect<StateProps, DispatchProps, OwnProps>(
   (state, ownProps) => ({
     addTokenPending: state.ui.wallets.addTokenPending,
-    wallet: state.ui.wallets.byId[ownProps.walletId]
+    wallet: state.ui.wallets.byId[ownProps.walletId],
+    currencyWallet: state.core.account.currencyWallets[ownProps.walletId],
+    currencyInfos: [] // TODO: replace it with real data when it will be available
```
Comment (thehobbit85) [inline src/components/scenes/AddTokenScene.js:254]: Select better name

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
PR 1:
Add new string constants related to Add Token scene
Add "defaultBorderColor" property to app theme
Add "fontFaceMedium" property to app theme and replace in all places … 
Add additional functions to the utils allowing to get ready edge-styles
Add the abilities to resize edges and display text as a warning in th…
Remove:
1.Add the ability for the SecondaryButton to display bold text
PR 2:
Add the ability for the SceneHeader to display bold text 
Add the ability for the ModalTitle to display bold text
Instead of those 2 PR’s, everything should just be default bold, and if any scene would require none default, we would fix it later in a !fixup commit.
PR 3:
Update Fade component: use pointerEvents instead of removing children…  
Add fade animation hook
Update Fade component to use fade animation hook
PR 4:
Add Checkbox component to the repo - Fix types based on this on in the login-ui:
https://github.com/EdgeApp/edge-login-ui/blob/e7fd8dc80d1ad66a0f5ab9a89f64450191444e13/packages/edge-login-ui-rn/src/components/themed/Checkbox.js
PR 5:
Add MismatchTokenParamsModal component to the repo - Remove “bolds”
PR 6:
Redesign AddToken scene
Remove AddToken title from main router render

**Pull request closed without merging**